janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/GhostVertexRemover.java

```diff
@@ -55,52 +54,56 @@
     private Instant jobStartTime;
 
     public GhostVertexRemover(JanusGraph graph) {
-        super(graph, new NoOpJob());
+        super(graph);
     }
 
     public GhostVertexRemover() {
-        this((JanusGraph)null);
+        this((JanusGraph) null);
     }
 
-    protected GhostVertexRemover(GhostVertexRemover copy) { super(copy); }
+    protected GhostVertexRemover(GhostVertexRemover copy) {
+        super(copy);
+    }
 
     @Override
-    public GhostVertexRemover clone() { return new GhostVertexRemover(this); }
-
+    public GhostVertexRemover clone() {
+        return new GhostVertexRemover(this);
+    }
 
     @Override
     public void workerIterationStart(Configuration jobConfig, Configuration graphConfig, ScanMetrics metrics) {
-        super.workerIterationStart(jobConfig, graphConfig, metrics);
-        Preconditions.checkArgument(jobConfig.has(GraphDatabaseConfiguration.JOB_START_TIME),"Invalid configuration for this job. Start time is required.");
-        this.jobStartTime = Instant.ofEpochMilli(jobConfig.get(GraphDatabaseConfiguration.JOB_START_TIME));
```

Comment:
Looks like previously this line was executed after graph is initialized but before the transaction is started. Right now it is executed after the transaction is started. 
I.e. the first time `workerIterationStart` is called the `jobStartTime` will be `null`. Thus, during the execution of `startTransaction`, we will set `null` here `txb.commitTime(jobStartTime);` instead of current time. Any next execution of `workerIterationStart` will set `commitTime` of the previous transaction instead of current transaction.
It looks like a bug to me in this refactoring, but I didn't check all the refactored code yet, so I could miss something

Reply:
Great catch! 

> Any next execution of workerIterationStart will set commitTime of the previous transaction instead of current transaction.

This won't happen because this method is only called once.